views/js/chatControl.js

```diff
@@ -2,8 +2,44 @@ var chatActiveRoom = "general";
 var chatRefreshTimeMs = 30000;
 
 var refreshChat = function () {
-  document.getElementById("chat_messages").innerHTML = fetchChatHTML();
-  setChatScrollToBottom();
+  fetch('./chat/' + chatActiveRoom)
+    .then(function (response) {
+      return response.json();
+    })
+    .then(function (chatList) {
+      var chatHTML = "";
+      chatList.forEach(function (element) {
+        chatHTML += `<div class="
```

Comment:
Prefer to build this up as an array and then join the array, rather than a bunch of individual string addition operations.

Reply:
reason:

"string" + "string" may require allocating a new string and copying both contents. Doing this N times is potentially N allocations and copies, where the left-hand side is length C*N; total copying is O(N^2).

Doing this as a single `join(array)` can be a single allocation and copy, i.e. O(N).